stan/math/prim/mat/meta/vector_seq_view.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef STAN_MATH_PRIM_MAT_META_VECTOR_SEQ_VIEW_HPP
+#define STAN_MATH_PRIM_MAT_META_VECTOR_SEQ_VIEW_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <vector>
+
+namespace stan {
+  using std::vector;
+  using Eigen::Matrix;
+  using Eigen::Dynamic;
+
+  /**
+   * Provides a low-cost wrapper for situations where you either need an Eigen
+   * Vector or RowVector or a std::vector of them and you want to be agnostic
+   * between those two options. This is similar to scalar_seq_view but instead
+   * of being a sequence-like view over a scalar or seq of scalars, it's a
+   * sequence-like view over a vector or seq of vectors. Notably this version
+   * only allows std::vectors as the sequence type, since we would have
+   * difficulty figuring out which contained type was the container otherwise.
+   *
+   * @tparam T the underlying scalar type inside the underlying Vector
+   */
+  template <typename T>
+  class vector_seq_view { };
+
+  template <typename T>
+  class vector_seq_view<Matrix<T, Dynamic, 1> > {
+    public:
+      explicit vector_seq_view(const Matrix<T, Dynamic, 1>& m): m_(m) {}
+      int size() const {
+        return 1;
+      }
+      Matrix<T, Dynamic, 1> operator[](int /* i */) const {
```

Comment:
We need doc for everything in specialized classes.  Nothing is inherited from the base template class, so it needs all new doc.   We have not been good about this in the past, but are trying to be going forward.